From 8b207385e4875d2a2d3af2f5aec958e08e3976ea Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:12 -0700
Subject: [PATCH 1637/1852] v4l: xilinx: vpss-scaler: Fix warnings for UNINIT

commit 57f8c0a2d104016da1b3290394a3befb56b0236f from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the NULL_RETURNS type.

The actual warnings are:

For xscaler_parse_of():
Event var_decl: 	Declaring variable "dt_ppc" without initializer.
Event uninit_use: 	Using uninitialized value "dt_ppc".

For xv_hscaler_calculate_phases():
Event var_decl: 	Declaring variable "nr_rds" without initializer
Event uninit_use: 	Using uninitialized value "nr_rds"

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index 939c8b4b0ce8..0d8ccea5b0f4 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1019,7 +1019,7 @@ xv_hscaler_calculate_phases(struct xscaler_device *xscaler,
 	bool get_new_pix;
 	u64 phaseH;
 	u64 array_idx = 0;
-	int nr_rds;
+	int nr_rds = 0;
 	int nr_rds_clck;
 	unsigned int nphases = xscaler->max_num_phases;
 	unsigned int nppc = xscaler->pix_per_clk;
@@ -1825,7 +1825,7 @@ static int xscaler_parse_of(struct xscaler_device *xscaler)
 	struct device_node *ports;
 	struct device_node *port;
 	int ret;
-	u32 port_id, dt_ppc;
+	u32 port_id, dt_ppc = 0;
 
 	if (xscaler->cfg->flags & XSCALER_CLK_PROP) {
 		xscaler->aclk_axis = devm_clk_get(dev, "aclk_axis");
-- 
2.31.1

